### PR TITLE
Merge into main for release v0.15.0 (v0.14.13)

### PR DESCRIPTION
**Pull request by chrisdiao**
* v0.14.1 | [Snyk] Security upgrade python from 3.10.0b1-alpine3.12 to rc-alpine3.12 [#571](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/571)
* v0.14.2 | updated threat model diagrams [#591](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/591)
* v0.14.3 | Update setup.sh [#567](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/567)
* v0.14.4 | Bdd test integration [#590](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/590)
* v0.14.5 | Implementation of Kibana dashboards autoload - issue 33 [#573](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/573)
* v0.14.6 | Properly handle bad SDK responses [#594](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/594)
* v0.14.7 | Get latest version of test_data when building an AMI image [#600](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/600)
* v0.14.8 | Utilize an older version of minio - one that works [#611](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/611)
* v0.14.9 | Removed verify=False in line 162 [#610](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/610)
* v0.14.10 | Revert "Removed verify=False in line 162" [#614](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/614)
* v0.14.11 | Change the SDK server to a working one [#625](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/625)
* v0.14.12 | Release 15 sonar cloud issue 620 [#627](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/627)
* v0.14.13 | Add cifs-utils to f2f plugin [#612](https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/612)

### CODE REVIEW

**Comment by ChrisHolman** [comment]
GitGuardian - Clean
Snyk SCA - Medium issues relating to requirements.txt 
Snyk Container - No containers identified
Snyk Infrastructure - Medium findings, affecting Helm, Terraform
SonarCloud - Critical - Enable server certificate validation on the SSL/TLS connection. Affecting the configure_Env.py file

-----

SonarCloud issue needs rectifying prior to this being merged. Configure_Env.py file, line 162. 

response = requests.request("GET", url , verify=False, timeout=10) 

Verify should be set to True, to enable validation on the connection. 

Reference: https://sonarcloud.io/project/issues?id=filetrust_cdr-plugin-folder-to-folder&open=AXqkhFneIcI4eL9d3Nkz&resolved=false&types=VULNERABILITY

**Comment by chrisdiao** [comment]
@DinisCruz FYI regarding the SPIRA result for this release
image

image

Thank you @MaNe2020 !

File: cdr_plugin_folder_to_folder/common_settings/Config.py [hunk 1]
```diff
@@ -26,7 +26,8 @@
 DEFAULT_KIBANA_HOST      = "kib01"
 DEFAULT_KIBANA_PORT      = "5601"
 DEFAULT_THREAD_COUNT     = 10
-DEFAULT_TEST_SDK         = 'gw-cloud-sdk-455649808.eu-west-1.elb.amazonaws.com'
+DEFAULT_TEST_SDK         = 'ec2-63-34-10-90.eu-west-1.compute.amazonaws.com'
```
Comment (DinisCruz) [inline cdr_plugin_folder_to_folder/common_settings/Config.py:29]: humm doesn't https://cloud-sdk.glasswallsolutions.com/ work?

File: cdr_plugin_folder_to_folder/processing/File_Processing.py [hunk 2]
```diff
@@ -257,6 +258,26 @@ def do_rebuild(self, endpoint, hash, source_path, dir):
...
+    def handleBadResponse(self, dir, response):
+        message = f"Not a regular SDK response"
+        if response.status_code != HTTPStatus.OK:
+            message = f"{response.status_code} {response.reason}."
+
+        try:
+            error_json_file_path = os.path.join(dir, "error.json")
+
+            with open(error_json_file_path, 'wb') as file:
+                file.write(response.content)
```
Comment (DinisCruz) [inline cdr_plugin_folder_to_folder/processing/File_Processing.py:270]: use helper file IO functions

File: cdr_plugin_folder_to_folder/utils/_to_refactor/For_OSBot_Elastic/Dashboard_Manager.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    def __init__(self):
+        self.config = Config()
+        self.host   = self.config.kibana_host
+        self.port   = self.config.kibana_port
+        self.kibana = Kibana(host=self.host, port=self.port).setup()
+        self.object_type    = 'dashboard_manager'
+        pass
+
+    def isDashbooard(self, filepath):
```
Comment (DinisCruz) [inline cdr_plugin_folder_to_folder/utils/_to_refactor/For_OSBot_Elastic/Dashboard_Manager.py:22]: should be `is_dashboard`

**Review by DinisCruz — APPROVED** [review]

**Comment by DinisCruz** [comment]
Approved for merge, although it is not a good sign that the code coverage was decreased

File: cdr_plugin_folder_to_folder/common_settings/Config.py (hunk 1 above)
Comment (ggrig) [reply cdr_plugin_folder_to_folder/common_settings/Config.py:29]: We still do not support HTTPS in F2F plugin (although that should not be a big deal)
And unfortunately, cloud-sdk.glasswallsolutions.com does not rebuild with HTTP requests

File: cdr_plugin_folder_to_folder/processing/File_Processing.py (hunk 2 above)
Comment (chrisdiao) [reply cdr_plugin_folder_to_folder/processing/File_Processing.py:270]: Created a [task](https://github.com/filetrust/cdr-plugin-folder-to-folder/issues/642) for this feedback @DinisCruz

Comment (ggrig) [reply cdr_plugin_folder_to_folder/processing/File_Processing.py:270]: Completely agree
We tried to resolve similar issues in https://github.com/filetrust/cdr-plugin-folder-to-folder/pull/602/files

File: cdr_plugin_folder_to_folder/utils/_to_refactor/For_OSBot_Elastic/Dashboard_Manager.py (hunk 3 above)
Comment (ggrig) [reply cdr_plugin_folder_to_folder/utils/_to_refactor/For_OSBot_Elastic/Dashboard_Manager.py:22]: Makes sense. 
Surprisingly SonarCloud has not catch it

Comment (chrisdiao) [reply cdr_plugin_folder_to_folder/utils/_to_refactor/For_OSBot_Elastic/Dashboard_Manager.py:22]: Created a [task](https://github.com/filetrust/cdr-plugin-folder-to-folder/issues/643) for this feedback @DinisCruz

**Pull request merged**